### ISSUE TITLE
Empty tooltips on hover

### ISSUE THREAD

**Issue by laughedelic** [scalameta/language-server#27]
#### Examples

* <img width="370" alt="screen shot 2017-11-10 at 03 03 32" src="https://user-images.githubusercontent.com/766656/32639356-d194a306-c5c3-11e7-82a2-ea653d7627d8.png">
* <img width="321" alt="screen shot 2017-11-10 at 00 29 21" src="https://user-images.githubusercontent.com/766656/32639266-5513c550-c5c3-11e7-8706-f9ea6b23ff46.png">
* <img width="342" alt="screen shot 2017-11-10 at 03 01 10" src="https://user-images.githubusercontent.com/766656/32639353-ce6422c4-c5c3-11e7-87bb-9c093ecaaf48.png">

#### File content

```scala
package example
import a._

class UserTest extends org.scalatest.FunSuite {

  val user = a.User("Bob", 42)

  val list = List(1, 2, 3).map(x => x.toString)

}
```

#### Log

<details><summary>...</summary>

```
03:05:13.798 DEBUG l.core.MessageReader - Received headers:
Content-Length: 242
03:05:13.798 DEBUG l.core.Connection - Received {"jsonrpc":"2.0","id":228,"method":"textDocument/hover","params":{"textDocument":{"uri":"file:///Users/laughedelic/dev/laughedelic/language-server/test-workspace/a/src/test/scala/example/UserTest.scala"},"position":{"line":8,"character":15}}}
03:05:13.799 INFO  s.m.l.ScalametaLanguageServer - hover at a/src/test/scala/example/UserTest.scala:8:15
03:05:13.799 INFO  s.m.l.ScalametaLanguageServer - Database for a/src/test/scala/example/UserTest.scala
03:05:13.799 INFO  s.m.l.ScalametaLanguageServer - _root_.example. at a/src/test/scala/example/UserTest.scala:0:8
03:05:13.799 INFO  s.m.l.ScalametaLanguageServer - _root_.a. at a/src/test/scala/example/UserTest.scala:2:7
03:05:13.799 INFO  s.m.l.ScalametaLanguageServer - _root_.example.UserTest# at a/src/test/scala/example/UserTest.scala:4:6
03:05:13.799 INFO  s.m.l.ScalametaLanguageServer - _root_.example.UserTest#`<init>`()V. at a/src/test/scala/example/UserTest.scala:4:15
03:05:13.799 INFO  s.m.l.ScalametaLanguageServer - _root_.org. at a/src/test/scala/example/UserTest.scala:4:23
03:05:13.799 INFO  s.m.l.ScalametaLanguageServer - _root_.org.scalatest. at a/src/test/scala/example/UserTest.scala:4:27
03:05:13.799 INFO  s.m.l.ScalametaLanguageServer - _root_.org.scalatest.FunSuite# at a/src/test/scala/example/UserTest.scala:4:37
03:05:13.799 INFO  s.m.l.ScalametaLanguageServer - _root_.example.UserTest#user. at a/src/test/scala/example/UserTest.scala:6:6
03:05:13.799 INFO  s.m.l.ScalametaLanguageServer - _root_.a. at a/src/test/scala/example/UserTest.scala:6:13
03:05:13.799 INFO  s.m.l.ScalametaLanguageServer - _root_.a.User. at a/src/test/scala/example/UserTest.scala:6:15
03:05:13.799 INFO  s.m.l.ScalametaLanguageServer - _root_.example.UserTest#list. at a/src/test/scala/example/UserTest.scala:8:6
03:05:13.799 INFO  s.m.l.ScalametaLanguageServer - _root_.scala.collection.immutable.List. at a/src/test/scala/example/UserTest.scala:8:13
03:05:13.800 DEBUG l.core.MessageWriter - Content-Length: 158



{"jsonrpc":"2.0","result":{"contents":[{"language":"scala","value":""}],"range":{"start":{"line":8,"character":13},"end":{"line":8,"character":17}}},"id":228}
03:05:13.800 DEBUG l.core.MessageWriter - payload: {"jsonrpc":"2.0","result":{"contents":[{"language":"scala","value":""}],"range":{"start":{"line":8,"character":13},"end":{"line":8,"character":17}}},"id":228}
```

</details>

**Comment by gabro** [comment]
Two things:

- the fact that an empty tooltip (as opposed to no tooltip) is displayed when no hover info is returned by the server is up to the client (vscode in this case). There's a related discussion here: https://github.com/Microsoft/language-server-protocol/issues/261

- the fact that those info are missing is a limitation of the semanticdb information we have. I'm experimenting with a different approach in #21

**Comment by laughedelic** [comment]
Thanks for the link to discussion in the LSP repo. But the problem here is that the response is neither `[]` nor `""`, it's `[""]`:

```
"contents":[{"language":"scala","value":""}]
```

And it corresponds to this line:

https://github.com/laughedelic/language-server/blob/ba7d552904b08601c86fd42b5ea1160c35d8636f/metaserver/src/main/scala/scala/meta/languageserver/ScalametaLanguageServer.scala#L257

So I added a check for non-empty `denotation.signature` and it seems to work well, both in VS Code and in Atom: `[]` is interpreted as no hover information and the tooltip is not shown 👌

**Comment by laughedelic** [comment]
Fixed by #21
